python/tests/test_df.py

```diff
@@ -0,0 +1,98 @@
+import unittest
```

Comment:
Out of curiosity: Why not `pytest`?

Reply:
it comes with python, so no need to install other stuff. But no feelings here; we can refactor this whole thing. =)